Change developer tier to serverless tier (#10824)
pingcap/docs: tidb-cloud/connect-to-tidb-cluster.md

+ Connect via a SQL client

>     - [Connect via standard connection](#connect-via-standard-connection): The standard connection exposes a public endpoint with traffic filters, so you can connect to your TiDB cluster from your laptop. For [Dedicated Tier](/tidb-cloud/select-cluster-tier.md#dedicated-tier), you can connect to your TiDB cluster using TLS, which ensures the security of data transmission from your applications to TiDB clusters.

Reviewer: As serverless Tier users also need to use TLS now, can we remove the "For [Dedicated Tier](/tidb-cloud/select-cluster-tier.md#dedicated-tier), you can connect to your TiDB cluster using TLS, which ensures the security of data transmission from your applications to TiDB clusters." sentence here?

Reviewer 2: I think yes.

Reviewer 3: How about this?
```suggestion
    - [Connect via standard connection](#connect-via-standard-connection): The standard connection exposes a public endpoint with traffic filters, so you can connect to your TiDB cluster from your laptop. This connection method enables TLS, which ensures the security of data transmission from your applications to TiDB clusters.
```
